server/content-repo/content-delivery.router.js

```diff
@@ -0,0 +1,43 @@
+const { ContentRepo, Enrollment } = require('../common/database');
+const { createError } = require('../common/errors');
+const ctrl = require('./content-delivery.controller');
+const HttpStatus = require('http-status');
+const router = require('express').Router();
+
+const { NOT_FOUND, UNAUTHORIZED } = HttpStatus;
+
+router
+  .use('/*', hasAccess)
+  .get('/', ctrl.list)
+  .get('/:contentId', ctrl.get)
+  .use('/:contentId/*', getRepoSourceId)
+  .get('/:contentId/container/:containerId', ctrl.getContainer)
+  .get('/:contentId/exam/:examId', ctrl.getExam)
+  .get('/:contentId/assessments/:assessmentsId', ctrl.getAssessments);
+
+function hasAccess({ cohort, user }, res, next) {
+  if (user.isAdmin()) return next();
+  const opts = { where: { studentId: user.id, cohortId: cohort.id } };
+  return Enrollment.findOne(opts)
+    .then(enrollment => {
+      if (enrollment) return next();
+      return createError(UNAUTHORIZED, 'Access restricted');
+    });
+}
+
+function getRepoSourceId(req, res, next) {
```

Comment:
Please rename it to getRepo and fetch other core data and then attach it to req.